src/Statement/StatementList.php

```diff
@@ -282,4 +282,24 @@ public function isEmpty() {
 		return empty( $this->statements );
 	}
 
+	/**
+	 * @since 3.0
+	 *
+	 * @param Statement $statement
+	 *
+	 * @return int|bool
+	 */
+	public function indexOf( Statement $statement ) {
+		$index = 0;
+
+		foreach ( $this->statements as $s ) {
+			if ( $s->equals( $statement ) ) {
```

Comment:
The only reason I'm adding this is because we need it in Wikibase.git. So it does make sense to look at what is actually needed there. If we create something else, then we're not solving the lack of having this functionality in non-deprecated code.